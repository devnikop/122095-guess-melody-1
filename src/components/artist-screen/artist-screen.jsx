import { connect } from "react-redux";
import PropTypes from "prop-types";
import React from "react";

import { ActionCreator } from "../../reducer/game/game";
import { artistQuestionType } from "../../types/types";

const ArtistScreen = ({
  question,
  onChange,
  renderAnswer,
}) => {
  const { answers, song } = question;

  const _handleInputChange = (index) => {
    onChange(index);
  }

  return (
    <section className="game game--artist">
      <section className="game__screen">
        <h2 className="game__title">Кто исполняет эту песню?</h2>
        <div className="game__track">
          {renderAnswer(song.src, 0)}
        </div>

        <form className="game__artist">
          {answers.map((answer, index) => (
            <div className="artist" key={index}>
              <input
                className="artist__input visually-hidden"
                id={`answer-${index}`}
                name="answer"
                type="radio"
                value={`answer-${index}`}
                onChange={_handleInputChange.bind(null, index)}
              />
              <label className="artist__name" htmlFor={`answer-${index}`}>
                <img
                  className="artist__picture"
                  alt={answer.artist}
                  src={answer.picture}
                />
                {answer.artist}
              </label>
            </div>
          ))}
        </form>
      </section>
    </section>
  );
}

ArtistScreen.propTypes = {
  question: artistQuestionType,
  onChange: PropTypes.func.isRequired,
  renderAnswer: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  onChange: ActionCreator.checkArtistQuestion
};

export { ArtistScreen };
export default connect(null, mapDispatchToProps)(ArtistScreen);
