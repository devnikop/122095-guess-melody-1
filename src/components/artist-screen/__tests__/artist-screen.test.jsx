import React from "react";
import renderer from "react-test-renderer";

import { ArtistScreen } from "../artist-screen.jsx";

const mock = {
  question: {
    answers: [
      {
        picture: `source`,
        artist: `John`,
      },
      {
        picture: `source2`,
        artist: `Jim`,
      },
    ],
    song: {
      artist: `artist`,
      src: `source`,
    },
    type: `artist`,
  },
};

const createNodeMock = (element) => {
  if (element.type === `audio`) {
    return {};
  }
};

it(`snapshot`, () => {
  const { question } = mock;

  const tree = renderer.create(
    <ArtistScreen onChange={jest.fn()} question={question} />,
    { createNodeMock }
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
