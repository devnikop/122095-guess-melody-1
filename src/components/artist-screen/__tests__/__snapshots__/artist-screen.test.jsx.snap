// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<section
  className="game game--artist"
>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    />
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="John"
            className="artist__picture"
            src="source"
          />
          John
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Jim"
            className="artist__picture"
            src="source2"
          />
          Jim
        </label>
      </div>
    </form>
  </section>
</section>
`;
